app/cdap/components/DrawerWidget/index.tsx

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import { Box, Container, Drawer, Typography } from '@material-ui/core';
+import grey from '@material-ui/core/colors/grey';
+import ChevronLeftRoundedIcon from '@material-ui/icons/ChevronLeftRounded';
+import CloseRoundedIcon from '@material-ui/icons/CloseRounded';
+import { Underline } from 'components/DrawerWidget/IconStore/Underline';
+import React, { ReactNode } from 'react';
+import styled from 'styled-components';
+
+export interface IDrawerWidgetProps {
+  headingText?: ReactNode | string;
+  openDrawer?: boolean;
+  showDivider?: boolean;
+  headerActionTemplate?: ReactNode | JSX.Element;
+  children?: ReactNode;
+  closeClickHandler?: () => void;
+  showBackIcon?: boolean;
+  anchor?: 'bottom' | 'left' | 'right' | 'top';
+}
+
+const BackIcon = styled(ChevronLeftRoundedIcon)`
+  font-size: 24px;
+  color: ${grey[600]};
+`;
+
+const CloseIcon = styled(CloseRoundedIcon)`
+  cursor: pointer;
+  font-size: 24px;
+`;
+
+const CustomizedDrawer = styled(Drawer)`
+& .MuiDrawer-paper {
+  top: 46px;
+  height: calc(100vh - 47px);
+}
+`;
+
+const Divider = styled.div`
+  width: 1px;
+  height: 28px;
+  background-color: #DADCE0;
+  margin: 0 15px;
+`;
+
+const DrawerContainer = styled(Container)`
+  width: 460px;
+  height: 100%;
+  padding-left: 30px;
+`;
+
+const DrawerHeader = styled.header`
+  height: 60px;
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  padding-left: 0px;
+  padding-right: 0px;
+`;
+
+const HeaderActions = styled(Box)`
+  display: flex;
+  align-items: center;
+`;
+
+const Label = styled(Typography)`
+  font-style: normal;
+  font-weight: 400;
+  font-size: 20px;
+  line-height: 150%;
+  letter-spacing: 0.15px;
+  color: #000000; /* Mui Colors not available */
+`;
+
+const LabelContainer = styled(Box)`
+  display: flex;
+  flex-direction: column;
+  align-items: flex-start;
+`;
+
+export default function ({
+  headingText,
+  openDrawer,
+  showDivider,
+  headerActionTemplate,
+  children,
+  closeClickHandler,
+  showBackIcon,
+  anchor,
```

Comment:
why not give this a default value? `anchor = 'right'`

Reply:
done